Aptoide/aptoide-client-v8
# [feature/ASV-1536] AppView migration promotion

**What does this PR do?**

   This generalizes promotions a bit better and specifically implements migration promotion and analytics. A new db object was needed do distinguish migrated apps from non-migrated ones.

**Database changed?**

   No

**Where should the reviewer start?**


**How should this be manually tested?**

  Check every main flow and see if it behaves as it should, some flows include:
  - Rewrite request for a package where two promotions are active but one of them is already claimed. In this case no promotion should appear.
  - Rewrite request for a package where two promotions are active and unclaimed. Further, the app should be ready to be migrated. In this case, the promotion should pertain to migration promotion. I think you'll have to check this in the debugger. Or maybe check claim requests
  - Rewrite request for a package where only migration promotion is active. In this case, promotion should only show to apps able to migrated.
  - Try to do a complete flow test as well (i.e. install wallet + migrate).
  - Test analytics again since they were refactored/implemented.
  - ... and any other case it may break :D 

**What are the relevant tickets?**

  [ASV-1536](https://aptoide.atlassian.net/browse/ASV-1536)

NOTE: Tests still missing...
**Code Review Checklist**
